### PR TITLE
Adding "socketPath" option for connection to database

### PR DESCRIPTION
**Pull request by saronoff**

### CODE REVIEW

**Comment by saronoff** [comment]
MySQL configuration allows connection using a socket

**Comment by saronoff** [comment]
I think the dbConfig has the database in the connection, which looks to be something this code wanted to avoid.

**Comment by kirrg001** [comment]
> I think the dbConfig has the database in the connection, which looks to be something this code wanted to avoid.

Correct. Either we keep this as is or we omit the database key.

**Comment by kirrg001** [comment]
With "as it is" i meant we keep the PR as is.

**Comment by acburdine** [comment]
There's also the option of doing something like:

```js
// at the top of the file
var omit = require('lodash/omit')

var connection = exports.connect(omit(dbConfig, ['dbkey']));
```

Not 100% sure what the dbConfig key is - but IMO using lodash's omit is the best approach as it allows any sort of database connection parameters to be passed, including something like a dsn connection string rather than database parameters.

**Comment by saronoff** [comment]
@kirrg001 Apologies for my misunderstanding 👍

**Comment by kirrg001** [comment]
@saronoff Could you please update the PR and use the code @acburdine suggested?
Then i will merge and release a new version 👍

**Comment by saronoff** [comment]
@kirrg001 I opened a new PR (#69) with a change based on the suggestion. Does that work? I figured I could do it without lodash since it was all pretty simple.

**Comment by kirrg001** [comment]
Personally i would prefer using `omit`, because it's simply shorter and we use `lodash` a lot in the code base anyway. Thank your so much for your effort 👋

File: lib/database.js [hunk 1]
```diff
@@ -36,17 +37,7 @@ exports.createDatabaseIfNotExist = function createDatabaseIfNotExist(dbConfig) {
...
-    var connection = exports.connect({
-        client: dbConfig.client,
-        connection: {
-            host: dbConfig.connection.host,
-            port: dbConfig.connection.port,
-            user: dbConfig.connection.user,
-            password: dbConfig.connection.password
-        }
-    });
+    var connection = exports.connect(omit(dbConfig, ['database']));
```
Comment (kirrg001) [inline lib/database.js]: `database` is a sub key of `connection`

Comment (saronoff) [inline lib/database.js]: That it is...one moment

**Pull request merged**